cookiecutter/repository.py

```diff
@@ -41,6 +41,20 @@ def expand_abbreviations(template, abbreviations):
     return template
 
 
+def valid_repository(repo_directory):
```

Comment:
I think we do a reasonable job with the [invalid json error message](https://github.com/audreyr/cookiecutter/blob/master/cookiecutter/generate.py#L85-L93), but I can certainly move that code to this function...